Debug information for file: javali_tests/HW3/ErrUnknownField1.javali
AST Resulting From Parsing Stage:
ClassDecl (name: X, superClass: Object)
| VarDecl (name: field, type: int)
ClassDecl (name: Main, superClass: Object)
| MethodDecl (argumentNames: [], argumentTypes: [], name: main, returnType: void)
| | Seq ()
| | | VarDecl (name: x, type: X)
| | Seq ()
| | | Assign ()
| | | | Var (name: x)
| | | | NewObject (typeName: X)
| | | Assign ()
| | | | Field (fieldName: field)
| | | | | Var (name: x)
| | | | IntConst (value: 0)
| | | Assign ()
| | | | Field (fieldName: notafield)
| | | | | Var (name: x)
| | | | IntConst (value: 1)
Error message from reference is: Type X has no field notafield
Error message: Assignment must have compatible types.

Phase semantic failed because we expected to see:
NO_SUCH_FIELD
But we actually saw:
TYPE_ERROR
The difference is:
At line   1 /   1:
< NO_SUCH_FIELD
> TYPE_ERROR

